src/ui/include/EditorNS/editor.h

```diff
@@ -307,6 +309,49 @@ namespace EditorNS
          */
         void requestCursorInfo();
 
+        /**
+         * @brief Get the current selection boundaries.
+         * @param callback  Lambda or functor.
+         *                  Must accept QList<Editor::Selection> type as parameter.
+         */
+        template<typename T>
+        void getSelections(T callback)
+        {
+            sendMessageWithCallback("C_FUN_GET_SELECTIONS", 
+                    [&, callback](const QVariant& v) {
+                callback(convertToSelections(v));
+            }); 
+        }
```

Comment:
I think that's near impossible for such a generic `T`. But we could move the function definition outside the class so we only have function declarations there.